src/Facebook/Monad.hs

```diff
@@ -99,6 +111,10 @@ data FbData = FbData
   { fbdCreds :: Credentials -- ^ Can be 'undefined'!
   , fbdManager :: !H.Manager
   , fbdTier :: !FbTier
+  , fbdApiVersion :: !ApiVersion
+  , fbdAppSecretProofAdder :: forall anyKind. Maybe (AccessToken anyKind)
```

Comment:
Is this type actually required ? Based on the value in the `Credential` type and presence of `Maybe (AccessToken anyKind)` in `fbreq` you can take the decision there whether to include the secret proof or not.

Reply:
@psibi, I tried first without the `fbdAppSecretProofAdder`, just doing the adding in `fbreq` as you said, but then I needed to get the credentials inside `fbreq`. But `getCredentials :: FacebookT Auth m a` breaks `fbreq :: FacebookT anyAuth m a`.